tests/unit/test_agw_network_configurator.py

```diff
@@ -0,0 +1,1194 @@
+            call("macaddress"),
+            call(":"),
+            call(" "),
+            call("02:45:b8:e6:23:c6"),
+            call("\n"),
+            call("      "),
+            call("set-name"),
+            call(":"),
+            call(" "),
+            call("eth0"),
+            call("\n"),
+            call("  "),
+            call("version"),
+            call(":"),
+            call(" "),
+            call("2"),
+            call("\n"),
+        ]
+        agw_network_configurator = AGWInstallerNetworkConfigurator(
+            self.TEST_NETWORK_INTERFACES,
+            None,
+            None,
+        )
+        with patch(
+            "magma_access_gateway_installer.agw_network_configurator.AGWInstallerNetworkConfigurator._network_interfaces_are_named_eth0_and_eth1",  # noqa: E501
+            new_callable=PropertyMock,
+            return_value=False,
+        ), patch(
+            "magma_access_gateway_installer.agw_network_configurator.AGWInstallerNetworkConfigurator._dns_configured",  # noqa: E501
+            new_callable=PropertyMock,
+            return_value=True,
+        ), patch(
+            "magma_access_gateway_installer.agw_network_configurator.AGWInstallerNetworkConfigurator._network_interfaces_config_files_exist",  # noqa: E501
+            new_callable=PropertyMock,
+            return_value=True,
+        ), patch(
+            "magma_access_gateway_installer.agw_network_configurator.AGWInstallerNetworkConfigurator._netplan_installed",  # noqa: E501
+            new_callable=PropertyMock,
+            return_value=False,
+        ), patch(
+            "magma_access_gateway_installer.agw_network_configurator.AGWInstallerNetworkConfigurator._networking_service_enabled",  # noqa: E501
+            new_callable=PropertyMock,
+            return_value=True,
+        ), patch(
+            "magma_access_gateway_installer.agw_network_configurator.AGWInstallerNetworkConfigurator._configure_grub",  # noqa: E501
+            Mock(),
+        ), patch(
+            "magma_access_gateway_installer.agw_network_configurator.AGWInstallerNetworkConfigurator._update_grub_cfg",  # noqa: E501
+            Mock(),
+        ), patch(
+            "magma_access_gateway_installer.agw_network_configurator.open",
+            mock_open(),
+        ) as mocked_open_file, patch(
+            "magma_access_gateway_installer.agw_network_configurator.yaml.safe_load",
+            Mock(return_value=mocked_cloud_init_content),
+        ):
+            agw_network_configurator.configure_network_interfaces()
+        mocked_open_file_calls = mocked_open_file.mock_calls
+        self.assertTrue(
+            call(agw_network_configurator.CLOUD_INIT_YAML_PATH, "r") in mocked_open_file_calls
+        )
+        self.assertTrue(
+            call(agw_network_configurator.CLOUD_INIT_YAML_PATH, "w") in mocked_open_file_calls
+        )
+        mocked_open_file().write.assert_has_calls(expected_cloud_init_file_write_calls)
```

Comment:
This validation makes more sense in terms of what I was saying in the earlier comments.